packages/jest-emotion/src/matchers.js

```diff
@@ -42,8 +42,64 @@ function toHaveStyleRule(
   received: *,
   property: *,
   value: *,
-  options?: { target?: string, media?: string } = {}
+  options?: { native?: boolean, target?: string, media?: string } = {}
 ) {
+  if (options.native) {
+    // received component is a React Native component
+    let style
+
+    if (typeof received.props === 'function') {
+      // the received component is rendered by enzyme shallow()
+      style = received
+        .dive()
+        .dive()
+        .dive()
+        .prop('style')
+    } else {
+      // the received component is rendered by native-testing-library
+      style = received.props.style
```

Comment:
is this guaranteed to be an array or an object? I recall RN styles being sometimes referenced by integers (produced by RN's `StyleSheet`), I'm not sure about this as I have only played a little bit with RN and I don't have any actual experience with it.

Reply:
I will double check. Thanks for the review!